sips/sip-013/sip-013-pox-upgrade.md

```diff
@@ -0,0 +1,373 @@
+# Preamble
+
+SIP Number: 013
+
+Title: Upgrading Proof-of-Transfer Consensus
+
+Author:
+    Aaron Blankstein <aaron@hiro.so>,
+    Jude Nelson <jude@stacks.org>
+
+Consideration: Technical
+
+Type: Consensus
+
+Status: Draft
+
+Created: 1 December 2021
+
+License: BSD 2-Clause
+
+Sign-off:
+
+Discussions-To: https://github.com/stacksgov/sips
+
+# Abstract
+
+This SIP proposes a set of updates to Stacking, the Proof-of-Transfer
+(PoX) consensus algorithm as implemented in the Stacks chain and
+originally proposed in [SIP-007](./sip-007-stacking-consensus.md).
+These updates improve the user and developer experience of
+participating in Stacking, add support for new behaviors in the
+Stacking smart contract, and address potential consensus challenges
+for PoX related chain reorganizations. In addition to proposing these
+changes, this SIP also outlines an approach for implementing these
+changes in the Stacks blockchain.
+
+# Introduction
+
+The current PoX implementation in the Stacks blockchain has several
+shortcomings for developers and users:
+
+1. After unlocking, Stackers **must** wait one cycle before Stacking
+   again. This is an implementation consequence, not a requirement of
+   the algorithm itself.
+2. If users fail to qualify during a cycle, their funds remain locked
+   for the cycle regardless.
+3. PoX cannot support advanced behaviors that could be used for
+   different kinds of smart contracts: changing the PoX reward
+   address, increasing or decreasing lock amounts, etc. These features
+   would enable new kinds of applications that build on top of PoX.
```

Comment:
I think there's a fourth shortcoming: nodes cannot determine which Stacks block is the PoX anchor block in reward cycle N without processing all Stacks blocks in all reward cycles up to N - 1.  This not only makes the initial chain sync tedious and error-prone, but also makes the chain vulnerable to deep reorgs by hiding PoX anchor blocks and precludes us from building light clients.  This fix is addressed by my PR here: https://github.com/blockstack/stacks-blockchain/pull/2707.

Reply:
Ah, nm -- I see you discuss that later on.